packages/oauth/oauth-provider/src/request/request-store-redis.ts

```diff
@@ -19,14 +19,18 @@ export class RequestStoreRedis implements RequestStore {
   }
 
   async readRequest(id: RequestId): Promise<RequestData | null> {
-    const data = await this.redis.get(id)
-    return data ? JSON.parse(data) : null
+    const value = await this.redis.get(id)
+    if (!value) return null
+
+    const data = decode(value)
+    if (!data) await this.redis.del(id)
+
+    return data
   }
 
   async createRequest(id: RequestId, data: RequestData): Promise<void> {
-    const timeFrame = data.expiresAt.getTime() - Date.now()
-    await this.redis.set(id, JSON.stringify(data), 'PX', timeFrame)
-    if (data.code) await this.redis.set(data.code, id, 'PX', timeFrame)
+    await this.redis.set(id, encode(data), 'PX', px(data.expiresAt))
+    if (data.code) await this.redis.set(data.code, id, 'PX', px(data.expiresAt))
```

Comment:
It is slightly awkward that `px(data.expiresAt)` will take a different value in each call here.  We could compute it once, maybe even additionally pipeline these two calls.

Reply:
adapted code to use absolute expiracy date instead